### DISCUSSION TITLE
Understanding priorities

### DISCUSSION THREAD

**Discussion by postyy2** [RakambdaOrg/ChannelPointsMiner#236] [General]
Hi @RakSrinaNa ,

I've been using your script and would like to start moving forward with testing advanced features.
How is the Priorities function used?

An example to see if I understand correctly.
I have a folder where all the streamers are. If I want to add one that always has priority to get points and stay on the IRC channel.

On **streamer.json** file will i need to write the following?

```
{
    "priorities": [
        {
          "type": "constant",
          "score": 99999
        }
    ]
}
```

And automatically the priority is set to the streamer or do I have to modify something in the config file?

**Comment by Rakambda** [comment]
Hey,

As a quick answer, yes.

---

Though there's 2 different things involved here:

* The channel points farming, which does uses the priorities
* The chat presence (IRC)

For the chat part, the easiest, there's no restrictions. You can join as many as you want. So the bot joins when a streamer starts streaming, and leaves when it ends. There's no notion of priorities or whatever.

I say there's no limit, but there may be one, but is in term of quantity of messages. I think if there's too many messages per x seconds, connection may be closed. So if lots of big streamers that you farm streams at the same time, it may cause issues, I don't the exact value though and may not be reached that easily, unless you join the chat of thousands of streamers.

---

Now for the points mining there's that notion of priorities. Each x amount of time, the bot will select the 2 online channels (as Twitch caps it to 2) that will be used to send your watching time. The way it is done is that for each channel, a score is computed, and the ones with the best scores will be picked.
The score calculations are made based on the `priorities` field you give in the config, and contains some triggers based on different states (subscription status, current points, drops available, ...).

If you want a channel to be always first, you can indeed give it a `constant` value that you know will always be higher than any possible values that will be computed for the other channels (so the 99999 seems to do the trick, if you use lower values elsewhere).

As an example, you can have this in your default streamer settings:
```json
"priorities": [
          {
            "type": "subscribed",
            "score": 1,
            "score2": 2,
            "score3": 3
          },
          {
            "type": "watchStreak",
            "score": 10
          },
          {
            "type": "drops",
            "score": 5
          }
        ]
```

What this does is highly prioritize a watch streak (to get the +50 of joining the stream), 
then drops (because drops) 
and then subscribe status based on its tier (as in the long run you'll get more points with the multiplier).
 
If all priorities are activated at once then we'd have a value < 20. So if we want another streamer to always have the priority, we can define him a constant priority like you put in your question. Though a value of 20 would be enough (99999 is cool too and I guess conveys a bit better the idea of "make it first pls" 😆 ).

One thing to be noted, you're overriding the default parameters for that streamer (your default priorities are gone), so either he'll be offline and not considered, or online and have a score of 99999. It doesn't really matter in that case as 99999 or 100010 (or whatever) won't make a difference, but in other cases it may be a thing to be careful of.

---

I don't know if it's understandable and answers your question or if I got lost in my explanations 😆 .

**Reply by Rakambda** [reply]
Maybe as an addition to "And automatically the priority is set".

Not sure what you mean by automatic. But if you put it in streamer.json, then only that streamer will have that priority rule (and will make it always picked when he's online).

If you mean something like "I just have to edit the file streamer.json and changes will be picked automatically", then it depends on your config. If you put something like `"reloadEvery": 30` in your main config file, then changes will be picked up in at most 30 minutes from when you did the changes without needing to restart the app (if you don't have it you'll have to restart the app to apply changes).
